src/common/components/CustomIcon/SchoolIcon/BaseIcon.tsx

```diff
@@ -0,0 +1,24 @@
+import { CustomIcon, type CustomIconProps } from "../CustomIcon";
+import { NTUIcon } from "@/common/components/CustomIcon/SchoolIcon/NTUIcon";
+import { NUSIcon } from "@/common/components/CustomIcon/SchoolIcon/NUSIcon";
+import { SMUIcon } from "@/common/components/CustomIcon/SchoolIcon/SMUIcon";
+
+export interface SchoolIconProps extends CustomIconProps {
+  school: "NTU" | "SMU" | "NUS";
+}
+
+export const SchoolIcon = (props: SchoolIconProps) => {
+  const { school, ...CustomIconProps } = props;

```

Comment:
Is there a reason for spreading the props here instead of in the function?

Reply:
not really, more of an oversight on my part! I have amended it